### ISSUE TITLE
Support regular expressions in include/exclude options

### ISSUE THREAD

**Issue by mlenkeit** [mateodelnorte/loop#33]
## 🚀 Feature Proposal

The `--include|exclude(-only)?` options only seem to support a list of directory names.

This feature proposal about adding support for regular expressions that are matched against the available sub directories in the CWD.

## Motivation

I have a set of repos in a directory and I want to loop over all web applications whose directory name starts with e.g. `angular-` to run a certain command, but I don't want to run it against any of the backend microservices. Especially when dealing with large projects where repos follow naming conventions, this would be very helpful.

## Example

```shell
$ loop --include angular-.*
# new option would be fine too, e.g.:
$ loop --include-pattern angular-.*
```

**Comment by mlenkeit** [comment]
Alright, I'll start working on a PR.

Do you have examples for some of the oddities that you are expecting? I will try to represent them in the tests. So far, I was assuming that this feature would be rather straight forward: take whatever expression is provided and filter the directories accordingly. If the expression is not escaped properly, it just wouldn't work.

Btw, I'd like this feature to be eventually available in the meta package, but it seems like this package is the proper place to implement this.

**Comment by mateodelnorte** [comment]
It will automatically be available in meta as soon as renovate bumps
versions.

If you enforce the regex must be in quotes, you may have no problems.
Otherwise! There are plenty of special characters in bash and zsh they
overlap with regular expressions.

**Comment by mlenkeit** [comment]
Closed with 03047f3a430bd19fda61b6407374e8bc662fdf14
